examples/platforms/cc2538/radio.c

```diff
@@ -390,22 +432,14 @@ void cc2538RadioProcess(otInstance *aInstance)
     if ((sState == kStateReceive && sReceiveFrame.mLength > 0) ||
         (sState == kStateTransmit && sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
     {
-#if OPENTHREAD_ENABLE_DIAG
-
-        if (otPlatDiagModeGet())
-        {
-            otPlatDiagRadioReceiveDone(aInstance, &sReceiveFrame, sReceiveError);
-        }
-        else
-#endif
+        // signal MAC layer for each received frame if promiscous is enabled
+        // otherwise only signal MAC layer for non-ACK frame
+        if (((HWREG(RFCORE_XREG_FRMFILT0) & RFCORE_XREG_FRMFILT0_FRAME_FILTER_EN) == 0) ||
+            (sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
         {
-            // signal MAC layer for each received frame if promiscous is enabled
-            // otherwise only signal MAC layer for non-ACK frame
-            if (((HWREG(RFCORE_XREG_FRMFILT0) & RFCORE_XREG_FRMFILT0_FRAME_FILTER_EN) == 0) ||
-                (sReceiveFrame.mLength > IEEE802154_ACK_LENGTH))
+            if (sReceiveDoneCallback)
             {
-
-                otPlatRadioReceiveDone(aInstance, &sReceiveFrame, sReceiveError);
+                sReceiveDoneCallback(aInstance, &sReceiveFrame, sReceiveError);
```

Comment:
@mszczodrak: These lines here are a good example of how this change improves the overall maintainability of the codebase.